jaxquantum/core/measurements.py

```diff
@@ -0,0 +1,149 @@
+""" Helpers. """
+
+from typing import List
+from jax import config, Array
+
+import jax.numpy as jnp
+from tqdm import tqdm
+
+from jaxquantum.core.qarray import Qarray, powm
+from jaxquantum.core.dims import Qtypes
+from jaxquantum.core.operators import identity, sigmax, sigmay, sigmaz
+
+config.update("jax_enable_x64", True)
+
+
+# Calculations ----------------------------------------------------------------
+
+def overlap(rho: Qarray, sigma: Qarray) -> Array:
+    """Overlap between two states or operators.
+
+    Args:
+        rho: state/operator.
+        sigma: state/operator.
+
+    Returns:
+        Overlap between rho and sigma.
+    """
+
+    if rho.isvec() and sigma.isvec():
+        return jnp.abs(((rho.to_ket().dag() @ sigma.to_ket()).trace())) ** 2
+    elif rho.isvec():
+        rho = rho.to_ket()
+        res = (rho.dag() @ sigma @ rho).data
+        return res.squeeze(-1).squeeze(-1)
+    elif sigma.isvec():
+        sigma = sigma.to_ket()
+        res = (sigma.dag() @ rho @ sigma).data
+        return res.squeeze(-1).squeeze(-1)
+    else:
+        return (rho.dag() @ sigma).trace()
+
+
+def fidelity(rho: Qarray, sigma: Qarray) -> float:
+    """Fidelity between two states.
+
+    Args:
+        rho: state.
+        sigma: state.
+
+    Returns:
+        Fidelity between rho and sigma.
+    """
+    rho = rho.to_dm()
+    sigma = sigma.to_dm()
+
+    sqrt_rho = powm(rho, 0.5)
+
+    return ((powm(sqrt_rho @ sigma @ sqrt_rho, 0.5)).tr()) ** 2
+
+
+def quantum_state_tomography(rho: Qarray, physical_basis: List, logical_basis:
```

Comment:
again, you may wish to accept a batched Qarray instead of a list, so that you can parallelize/batch the computations